### PR TITLE
Ignore Outliers in Java API Memory Test

### PR DESCRIPTION
**Pull request by dyastremsky**
- Ignore outliers when calculating memory growth
- Separate sanity check (checking accuracy) from longer-running memory growth test
- Add long-running stress test components to test script (will officially add to long-running stress test in future PR).

### CODE REVIEW

**Comment by dyastremsky** [comment]
@GuanLuo For the internal CI, should I wait until your branch is merged or can I add the long-running labels for this test on the main branch (once we merge Java API into main)?

**Review by GuanLuo — APPROVED** [review]

**Comment by GuanLuo** [comment]
I will include this test to the stress test branch once it is merged in and start a run there

**Comment by dyastremsky** [comment]
> I will include this test to the stress test branch once it is merged in and start a run there

Perfect, thanks!

**Pull request merged**
